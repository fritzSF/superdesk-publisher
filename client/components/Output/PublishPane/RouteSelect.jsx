import React, { Component } from "react";
import PropTypes from "prop-types";
import Store from "../Store";
import OptGroup from "../../UI/OptGroup";

const RouteSelect = (props) => {
  const store = React.useContext(Store);
  const pubConfig = store.config.publisher || {};

  let collectionRoutes = props.routes.filter(
    (route) => route.type === "collection"
  );
  let contentRoutes = props.routes.filter((route) => route.type === "content");
  let customRoutes = props.routes.filter(
    (route) => route.type != "content" && route.type != "collection"
  );

  return (
    <div className="sd-line-input sd-line-input--is-select sd-line-input--dark-ui sd-line-input--no-margin">
      <label className="sd-line-input__label">Route</label>
      <select
        name="routeId"
        className="sd-line-input__select"
        value={props.selectedRouteId}
        onChange={props.onChange}
      >
        {props.routes.length && !props.selectedRouteId && (
          <option value="" disabled>
            Select a route
          </option>
        )}

        {!props.routes.length && (
          <option value="" disabled>
            No routes defined
          </option>
        )}
        <OptGroup
          list={collectionRoutes}
          valueField="id"
          nameField="name"
          label="Collection"
        />
        {pubConfig.hideContentRoutesInPublishPane ? null : (
          <OptGroup
            list={contentRoutes}
            valueField="id"
            nameField="name"
            label="Content"
          />
        )}

        {pubConfig.hideCustomRoutesInPublishPane ? null : (
          <OptGroup
            list={customRoutes}
            valueField="id"
            nameField="name"
            label="Custom"
          />
        )}
      </select>
    </div>
  );
};

RouteSelect.propTypes = {
  onChange: PropTypes.func.isRequired,
  selectedRouteId: PropTypes.any.isRequired,
  routes: PropTypes.array.isRequired,
};

export default RouteSelect;
