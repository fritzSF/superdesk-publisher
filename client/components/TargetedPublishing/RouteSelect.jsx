import React, { Component } from "react";
import axios from "axios";
import PropTypes from "prop-types";

import OptGroup from "../UI/OptGroup";

class RouteSelect extends Component {
  constructor(props) {
    super(props);

    this.state = {
      routes: [],
      loading: true,
    };

    this.pubConfig = this.props.config.publisher || {};
  }

  componentDidMount() {
    this.getRoutes();
  }

  getRoutes = () => {
    axios
      .get(this.props.apiUrl + "content/routes/", {
        headers: this.props.apiHeader,
        params: { limit: 1000 },
      })
      .then((res) => {
        this.setState({
          routes: res.data._embedded._items,
          loading: false,
        });
      });
  };

  render() {
    let collectionRoutes = this.state.routes.filter(
      (route) => route.type === "collection"
    );
    let contentRoutes = this.state.routes.filter(
      (route) => route.type === "content"
    );
    let customRoutes = this.state.routes.filter(
      (route) => route.type != "content" && route.type != "collection"
    );

    return (
      <div className="sd-line-input sd-line-input--is-select sd-line-input--dark-ui sd-line-input--no-margin">
        <label className="sd-line-input__label">Route</label>
        <select
          name="routeId"
          className="sd-line-input__select"
          value={this.props.selectedRouteId}
          onChange={this.props.onChange}
        >
          {this.state.loading && (
            <option value="" disabled>
              Loading...
            </option>
          )}
          {this.state.routes.length && (
            <option value="" disabled>
              Select a route
            </option>
          )}
          {!this.state.routes.length && !this.state.loading && (
            <option value="" disabled>
              No routes defined
            </option>
          )}
          <OptGroup
            list={collectionRoutes}
            valueField="id"
            nameField="name"
            label="Collection"
          />

          {this.pubConfig.hideContentRoutesInPublishPane ? null : (
            <OptGroup
              list={contentRoutes}
              valueField="id"
              nameField="name"
              label="Content"
            />
          )}

          {this.pubConfig.hideCustomRoutesInPublishPane ? null : (
            <OptGroup
              list={customRoutes}
              valueField="id"
              nameField="name"
              label="Custom"
            />
          )}
        </select>
      </div>
    );
  }
}

RouteSelect.propTypes = {
  onChange: PropTypes.func.isRequired,
  selectedRouteId: PropTypes.any.isRequired,
  apiHeader: PropTypes.object.isRequired,
  apiUrl: PropTypes.string.isRequired,
  config: PropTypes.object.isRequired,
};

export default RouteSelect;
