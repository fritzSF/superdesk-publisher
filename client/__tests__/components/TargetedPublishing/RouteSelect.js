import React from "react";
import RouteSelect from "../../../components/TargetedPublishing/RouteSelect";
import { render, fireEvent, waitForElement } from "@testing-library/react";
import axios from "axios";

jest.mock("axios");

describe("TargetedPublishing/RouteSelect", () => {
  it("renders correctly", async () => {
    const { container, getByText } = render(
      <RouteSelect
        config={{ publisher: {} }}
        apiUrl="example.com/"
        apiHeader={{ Authrization: "Basic 1234567" }}
        onChange={jest.fn()}
        selectedRouteId={2}
      />
    );

    await waitForElement(() => getByText("route2"));

    expect(container.firstChild).toMatchSnapshot();
  });

  it("fires onChange", async () => {
    const onChange = jest.fn();
    const { container } = render(
      <RouteSelect
        config={{ publisher: {} }}
        apiUrl="example.com/"
        apiHeader={{ Authrization: "Basic 1234567" }}
        onChange={onChange}
        selectedRouteId={2}
      />
    );

    const select = container.querySelector('select[name="routeId"]');

    fireEvent.change(select, { target: { value: 1 } });
    expect(onChange).toHaveBeenCalled();
  });
});
