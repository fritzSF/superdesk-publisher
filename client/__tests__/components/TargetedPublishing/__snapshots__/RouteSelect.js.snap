// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetedPublishing/RouteSelect renders correctly 1`] = `
<div
  class="sd-line-input sd-line-input--is-select sd-line-input--dark-ui sd-line-input--no-margin"
>
  <label
    class="sd-line-input__label"
  >
    Route
  </label>
  <select
    class="sd-line-input__select"
    name="routeId"
  >
    <option
      disabled=""
      value=""
    >
      Select a route
    </option>
    <optgroup
      label="Custom"
      style="color: black;"
    >
      <option
        value="1"
      >
        route1
      </option>
      <option
        value="2"
      >
        route2
      </option>
    </optgroup>
  </select>
</div>
`;
